spec/at-rules/unknown.md

```diff
@@ -14,11 +16,51 @@ structures it allows. It uses the following grammar:
 No whitespace is allowed after `@`. As with all statements, an `UnknownAtRule`
 without a block must be separated from other statements with a semicolon.
 
-When an at-rule is executed, its name is evaluated to produce an unquoted string
-which is used as the name of the generated at-rule. Then that generated name is
-checked to see if it's an at-rule that has special runtime handling.
+## Semantics
+
+To execute an unknown at-rule `rule`:
+
+* Let `name` be the result of evaluating `rule`'s `InterpolatedIdentifier`.
+
+* Let `value` be the result of evaluating `rule`'s `InterpolatedValue`, if it
+  exists.
+
+* Let `css` be a CSS unknown at-rule with name `name`, value `value`, and with
+  an empty list of children if `rule` has `Statements`.
+
+* Let `parent` be the [current style rule], [keyframe block], or at-rule if one
+  exists; or the innermost if multiple exist.
+
+  [current style rule]: ../style-rules.md#current-style-rule
+  [keyframe block]: ../style-rules.md#current-style-rule
+
+* If `parent` is a keyframe block, throw an error.
+
+* If `rule` has `Statements`:
+
+  * If `parent` isn't set, append `css` to [the current module]'s CSS.
+
+  * Otherwise, if `parent` is a style rule:
+
+    * If `rule`'s name is "font-face", or if its [unprefixed] name is
+      "keyframes", append `css` to [the current module]'s CSS.
+
+    * Otherwise:
+
+      * Let `copy` by a copy of `parent` without any children.
+
+      * Append `copy` to `parent`'s parent.
+
+      * Append `css` to `copy`.
```

Comment:
The way I read these lines would mean that:
```scss
.foo {
  a: b;
  @c {
    d: e;
  }
}
```
becomes
```scss
.foo {
  a: b;
}
.foo { // `copy` appended to `parent`
  @c { // `css` appended to `copy`
    d: e; // `rule`'s Statements appended to `css`
  }
}
```
but in practice I get:
```scss
.foo {
  a: b;
}
@c { // `css` appended to `parent`
  .foo { // `copy` appended to `css`
    d: e; // `rule`'s Statements appended to `copy`
  }
}
```

Reply:
I think this needs to be something like...
```suggestion
      * Let `copy` by a copy of `parent` without any children.

      * Append `css` to `parent`'s parent.

      * Append `copy` to `css`.
      
      * Assign `copy` to `css`.
```

Reply:
Ah, you're right, thanks!